polyphemus/lib/etls/metis_files_linker_base.rb

```diff
@@ -161,8 +162,15 @@ def should_skip_record?(record_name)
     false
   end
 
-  def record_name_by_path(path, regex)
-    match = path.match(regex)
+  def record_name_by_path(file_or_folder)
+    if file_or_folder.is_a?(Etna::Clients::Metis::File)
+      match = file_or_folder.file_path.match(@record_name_regex)
+    elsif file_or_folder.is_a?(Etna::Clients::Metis::Folder)
+      match = (file_or_folder.folder_path + '/').match(@record_name_regex)
```

Comment:
So here's the trick -- the record_name_regex, to differentiate between folder segments and filenames (basenames), requires a '/'.  This helps prevent us from picking up files that happen to match record names but are not records.  In the case of a folder, when we're processing just the folder name itself, we artificially add the ending slash so the regex understands that it is a directory path segment and not a filename.